modules/apps/headless/headless-object/headless-object-test/src/testIntegration/java/com/liferay/headless/object/resource/v1_0/test/CollaboratorResourceTest.java

```diff
@@ -6,14 +6,572 @@
+
+		return testGroup.getGroupKey();
+	}
+
+	private static ObjectDefinition _getObjectDefinition() throws Exception {
+		return ObjectDefinitionTestUtil.publishObjectDefinition(
+			true, ObjectDefinitionTestUtil.getRandomName(),
+			Collections.singletonList(
+				new TextObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).indexed(
+					true
+				).indexedAsKeyword(
+					true
+				).name(
+					"title"
+				).localized(
+					false
+				).build()),
+			ObjectDefinitionConstants.SCOPE_SITE, TestPropsValues.getUserId());
+	}
+
+	private ObjectEntry _addObjectEntry() throws Exception {
+		return _objectEntryLocalService.addObjectEntry(
+			TestPropsValues.getUserId(), testGroup.getGroupId(),
+			_objectDefinition.getObjectDefinitionId(), 0, null,
+			HashMapBuilder.<String, Serializable>put(
+				"title", RandomTestUtil.randomString()
+			).build(),
+			ServiceContextTestUtil.getServiceContext(
+				testGroup.getGroupId(), TestPropsValues.getUserId()));
+	}
+
+	private ObjectEntryFolder _addObjectEntryFolder() throws Exception {
+		return _objectEntryFolderLocalService.addObjectEntryFolder(
+			null, TestPropsValues.getUserId(), testGroup.getGroupId(),
+			ObjectEntryFolderConstants.PARENT_OBJECT_ENTRY_FOLDER_ID_DEFAULT,
+			HashMapBuilder.put(
+				LocaleUtil.ENGLISH, RandomTestUtil.randomString()
+			).build(),
+			RandomTestUtil.randomString(), new ServiceContext());
+	}
+
+	private Collaborator _addUserCollaborator(long classNameId, long classPK)
+		throws Exception {
+
+		User user = _getUser();
+
+		return _toCollaborator(
+			_sharingEntryLocalService.addSharingEntry(
+				null, TestPropsValues.getUserId(), 0, user.getUserId(),
+				classNameId, classPK, testGroup.getGroupId(), true,
+				Arrays.asList(SharingEntryAction.VIEW), null,
+				ServiceContextTestUtil.getServiceContext(
+					testGroup.getGroupId(), TestPropsValues.getUserId())));
+	}
+
+	private Collaborator _addUserGroupCollaborator(
+			long classNameId, long classPK)
+		throws Exception {
+
+		UserGroup userGroup = _getUserGroup();
+
+		return _toCollaborator(
+			_sharingEntryLocalService.addSharingEntry(
+				null, TestPropsValues.getUserId(), userGroup.getUserGroupId(),
+				0, classNameId, classPK, testGroup.getGroupId(), true,
+				Arrays.asList(SharingEntryAction.VIEW), null,
+				ServiceContextTestUtil.getServiceContext(
+					testGroup.getGroupId(), TestPropsValues.getUserId())));
+	}
+
+	private User _getUser() throws Exception {
+		User user = UserTestUtil.addUser();
+
+		_users.add(user);
+
+		return user;
+	}
+
+	private Collaborator _getUserCollaborator() throws Exception {
+		User user = _getUser();
+
+		return new Collaborator() {
+			{
+				actionIds = new String[] {
+					SharingEntryAction.VIEW.getActionId()
+				};
+				dateExpired = _randomDatePlusAYear();
+				externalReferenceCode = user.getExternalReferenceCode();
+				name = user.getFullName();
+				share = true;
+				type = "user";
```

Comment:
@Mikellorza, can we have an enum in the yaml for this? So we can reference it

Reply:
I will try and I will come back here with the results

Reply:
I am afraid there is no enum field type to use in the yaml :( . The validation throws an error like the following: 'Schema validation: Value should be one of: "array", "boolean", "integer", "null", "number", "object", "string" '

Reply:
the enum we talked about

Reply:
done, I added it and the code now looks better, 

thanks mozo